test/integration/stomp_/publisher/create_request_test.py

```diff
@@ -43,47 +21,71 @@ def system_spec():
     }
 
 
-class MessageListener(object):
-    create_event_captured = False
+class MessageListener(ConnectionListener):
+    _create_event_captured: bool
+
+    def __init__(self):
+        super().__init__()
+        self._create_event_captured = False
 
     def on_error(self, headers, message):
         print("received an error %s" % headers)
 
-    def on_message(self, headers, message):
-        try:
-            if headers["model_class"] == "Operation":
+    def on_message(self, headers: Dict[str, Any], message: Any) -> None:
+        """Set a property when a MESSAGE frame is received by the STOMP connection.
 
-                parsed = SchemaParser.parse_operation(message, from_string=True)
+        If the needed key is in the headers dict and has the correct value, signal
+        that we've been successful.
 
-                if parsed.model and parsed.model.name:
-                    if parsed.model.name.startswith("REQUEST"):
-                        print(message)
-                        self.create_event_captured = True
+        Args:
+            headers: a dictionary containing all headers sent by the server
+            as key/value pairs.
+
+            message: the frame's payload - the message body.
+        """
+        try:
+            if "model_class" in headers:
+                if headers["model_class"] == "Operation":
+                    parsed = SchemaParser.parse_operation(message, from_string=True)
+
+                    if parsed.model and parsed.model.name:
+                        if parsed.model.name.startswith("REQUEST"):
+                            self.create_event_captured = True
+                        else:
+                            print(f"ERROR: no 'REQUEST' is parsed model")
                     else:
-                        print(parsed.model.name)
-            elif headers["model_class"] == "error_message":
-                print("Error Message Returned:", message)
-        except:
-            print("Error: unable to parse message:", message)
+                        print("ERROR: no parsed model found")
+                elif headers["model_class"] == "error_message":
+                    print(f"ERROR: Message returned: {message!r}")
+                else:
```

Comment:
I think this else statement and the corresponding print message are in one level too far based on your above stated goal.  That is, this will print if "model_class" is present, but isn't one of the value you're checking for.

Also, could you have achieved the same thing by just outputting the exception message in the Exception block?  That would give you the key error when "model_class" isn't present, but then also be useful if some other, unexpected exception is actually what gets thrown.

Reply:
Duly noted on the print.

As for you other point, I would have not written it this way, but I tried to preserve the intent and form of the original and add some extra guards I think are appropriate for the form it's in.